internal/topo/planner/planner.go

```diff
@@ -371,19 +375,70 @@ func createLogicalPlan(stmt *ast.SelectStatement, opt *api.RuleOption, store kv.
 		p.SetChildren(children)
 		children = []LogicalPlan{p}
 	}
-
 	if stmt.Fields != nil {
-		p = ProjectPlan{
-			fields:      stmt.Fields,
-			isAggregate: xsql.IsAggStatement(stmt),
-			sendMeta:    opt.SendMetaToSink,
-		}.Init()
-		p.SetChildren(children)
+		srfMapping, err := extractSRFFromFileds(stmt)
+		if err != nil {
+			return nil, err
+		}
+		if len(srfMapping) > 0 {
+			psp := ProjectSetPlan{
+				SrfMapping:  srfMapping,
+				ProjectPlan: &ProjectPlan{},
+			}
+			psp.fields = stmt.Fields
+			psp.isAggregate = xsql.IsAggStatement(stmt)
+			psp.sendMeta = opt.SendMetaToSink
+			p = psp.Init()
+			p.SetChildren(children)
+		} else {
+			p = ProjectPlan{
+				fields:      stmt.Fields,
+				isAggregate: xsql.IsAggStatement(stmt),
+				sendMeta:    opt.SendMetaToSink,
+			}.Init()
+			p.SetChildren(children)
+		}
 	}
-
 	return optimize(p)
 }
 
+func checkSRFINConditions(condition ast.Expr) error {
+	var err error
+	ast.WalkFunc(condition, func(n ast.Node) bool {
+		switch f := n.(type) {
+		case *ast.Call:
+			if f.FuncType == ast.FuncTypeSrf {
+				err = fmt.Errorf("set-returning-function is not supported in where conditions")
+				return false
+			}
+		}
+		return true
+	})
+	return err
+}
+
+// extractSRFFromFileds extracts the set-returning-functions in the select field.
+func extractSRFFromFileds(stmt *ast.SelectStatement) (map[string]struct{}, error) {
+	srfCount := 0
+	nonSrfCount := 0
+	srfMapping := make(map[string]struct{}, 0)
+	for _, field := range stmt.Fields {
```

Comment:
how about nest unnest function, like abs(unnest(a))

Reply:
good catch, we should validate that `srf` shouldn't be nested for now.